packages/design-system/src/components/SearchBar/SearchBar.js

```diff
@@ -0,0 +1,184 @@
+import * as React from 'react';
+import * as PropTypes from 'prop-types';
+import cx from 'classnames';
+import SearchIcon from 'react-material-icon-svg/dist/MagnifyIcon';
+import CloseIcon from 'react-material-icon-svg/dist/CloseIcon';
+import { debounce } from '@livechat/data-utils';
+
+import styles from './style.scss';
+import { Loader } from '../Loader';
+import FieldError from '../FieldError';
+import { Input } from '../InputField';
+import { KeyCodes } from '../../constants/keyCodes';
+
+const acceptedSizes = ['basic', 'compact'];
+const baseClass = 'search-bar';
+const noop = () => {};
+
+class SearchBar extends React.PureComponent {
+  constructor(props) {
+    super(props);
+
+    this.inputRef = React.createRef();
+    this.debouncedOnChange = debounce(this.props.debounceInMs, value => {
+      this.props.onChange(value);
+    });
+
+    this.state = {
+      searchTerm: '',
+      isInCompactMode: this.props.size === 'compact'
+    };
+  }
+
+  handleChange = e => {
+    const { value } = e.target;
+    const { debounceInMs, onChange } = this.props;
+
+    this.setState({
+      searchTerm: value
+    });
+
+    if (debounceInMs) {
+      this.debouncedOnChange(value);
+    } else {
+      onChange(value);
+    }
+  };
+
+  handleKeyPress = key => {
+    const { onSubmit } = this.props;
+
+    if (onSubmit) {
+      if (key.keyCode === KeyCodes.enter) {
+        onSubmit(this.state.searchTerm);
+      }
+    }
+  };
+
+  handleClear = () => {
+    const { onChange, onSubmit } = this.props;
+    const searchFunction = onSubmit || onChange;
+
+    this.setState({
+      searchTerm: ''
+    });
+
+    // 1) Cancel debounced search
+    // 2) Use onClear instead of onSubmit | onChange function
+    searchFunction('');
+
+    if (this.inputRef.current) {
+      this.inputRef.current.focus();
+    }
+  };
+
+  handleCloseIconKeyDown = key => {
+    if (key.keyCode === KeyCodes.enter) {
+      this.handleClear();
+    }
+  };
+
+  toggleCompactMode = () => {
+    this.setState(
+      prevState => ({
+        isInCompactMode: !prevState.isInCompactMode
+      }),
+      () => {
+        if (!this.state.isInCompactMode && this.inputRef.current) {
+          this.inputRef.current.focus();
+        }
+      }
+    );
+  };
+
+  render() {
+    const { className, placeholder, value, loading, size, error } = this.props;
+    const { searchTerm, isInCompactMode } = this.state;
+
+    const isCompactSize = size === 'compact';
+    const shouldDisplayCloseButton = searchTerm && !loading && !isInCompactMode;
+
+    return (
+      <span className={className}>
```

Comment:
div in span is probably not a good idea. try using https://validator.w3.org/nu/#textarea. It will throw an error: `Error: Element div not allowed as child of element span in this context. (Suppressing further errors from this subtree.)`